experimental/unicodeset_parser/src/parse.rs

```diff
@@ -169,6 +184,47 @@ impl ParseError {
     }
 }
 
+/// The value of a variable in a UnicodeSet. Used as value type in [`VariableMap`].
+#[derive(Debug, Clone)]
+#[non_exhaustive]
+pub enum VariableValue<'a> {
+    /// A UnicodeSet, represented as a [`CodePointInversionListAndStringList`](CodePointInversionListAndStringList).
+    UnicodeSet(CodePointInversionListAndStringList<'a>),
+    // in theory, a one-code-point string is always the same as a char, but we might want to keep
+    // this variant for efficiency?
+    /// A single code point.
+    Char(char),
+    /// A string.
+    String(String),
+}
+
+impl From<char> for VariableValue<'_> {
```

Comment:
Prefer explicit construction over `.into()` calls, as it often makes callsites harder to read. Instead maybe add three convenience methods to `VariableMap`: `insert_set`, `insert_char`, `insert_string`.